src/o365/graph/commands/schemaextension/schemaextension-add.ts

```diff
@@ -0,0 +1,244 @@
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/schemaExtensions`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none',
+            'content-type': 'application/json'
+          }),
+          body: {
+            id: args.options.id,
+            description: args.options.description,
+            owner: args.options.owner,
+            targetTypes,
+            properties
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: any): void => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        cmd.log(`Schema extension added with Id '${args.options.id}'`);
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --id <id>',
+        description: `The unique identifier for the schema extension definition`
+      },
+      {
+        option: '-d, --description [description]',
+        description: 'Description of the schema extension'
+      },
+      {
+        option: '-o, --owner <owner>',
+        description: `The Id of the Azure AD application that is the owner of the schema extension`
+      },
+      {
+        option: '-t, --targetTypes <types>',
+        description: `Comma-separated list of Microsoft Graph resource types the schema extension targets`
+      },
+      {
+        option: '-p, --properties <properties>',
+        description: `The collection of property names and types that make up the schema extension definition formatted as a JSON string`
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.id) {
+        return 'Required option id is missing';
+      }
+
+      if (!args.options.owner) {
+        return 'Required option owner is missing';
+      }
+
+      if (args.options.owner && !Utils.isValidGuid(args.options.owner)) {
+        return `The specified owner '${args.options.owner}' is not a valid App Id`;
+      }
+
+      if (!args.options.targetTypes) {
+        return 'Required option targetTypes is missing';
+      }
+
+      if (!args.options.properties) {
+        return 'Required option targetTypes is missing';
+      }
+
+      if (!Utils.isValidJsonString(args.options.properties)) {
```

Comment:
With this approach, we're parsing the JSON string multiple times. It's inefficient and we should look into changing the code so that we only need to parse JSON once

Reply:
Refactored it to parse it only once, returing isValid and the parsed object